### ISSUE TITLE
Akka.Persistence journal fails to start in tests

### ISSUE THREAD

**Issue by kantora** [akkadotnet/akka.net#2026]
I've found the strange problem. In tests if I set `akka.test.calling-thread-dispatcher` as default dispatcher the default `inmem` journal (along with any other) will fail to start.

Even if I use the default dispatcher but will send to journal some custom message for example like this:
`Persistence.Instance.CreateExtension((ExtendedActorSystem)this.Sys).JournalFor(null).Tell("hello")` as the first test instruction it would fail to start.

I see this in logs:

```
2016-06-01 11:25:41.963 +03:00 [Debug] "Started (Akka.TestKit.Internal.InternalTestActor)"
2016-06-01 11:25:42.017 +03:00 [Debug] "now supervising akka://test/system/akka.persistence.journal.inmem"
2016-06-01 11:25:42.036 +03:00 [Error] "Object reference not set to an instance of an object."
[akka://test/system/akka.persistence.journal.inmem]: Akka.Actor.ActorInitializationException: Exception during creation ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Akka.Actor.Props.NewActor()
   at Akka.Actor.ActorCell.CreateNewActorInstance()
   at Akka.Actor.ActorCell.<>c__DisplayClass113_0.<NewActor>b__0()
   at Akka.Actor.ActorCell.UseThreadContext(Action action)
   at Akka.Actor.ActorCell.NewActor()
   at Akka.Actor.ActorCell.Create(Exception failure)
   --- End of inner exception stack trace ---
   at Akka.Actor.ActorCell.Create(Exception failure)
   at Akka.Actor.ActorCell.SystemInvoke(Envelope envelope)
```

**Comment by gengle** [comment]
Is there more context on this bug?  I have the same issue with testkit 1.2 & akka.persistence 1.2.0.36-beta

Here's a failing test.

```
    [TestFixture]
    public class PersistenceTestFixture : TestKit
    {
        [Test]
        public void inmem_should_not_crash()
        {
            var sut = Sys.ActorOf(Props.Create<FooActor>(), "foo1");

            sut.Tell("hello, why am I failing?");

            Assert.IsNotNull(sut);
            // [ERROR][6/9/2017 10:35:28 PM][Thread 0011][akka://test/system/akka.persistence.journal.inmem] Object reference not set to an instance of an object.
            // Cause: [akka://test/system/akka.persistence.journal.inmem#1974987499]: Akka.Actor.ActorInitializationException: Exception during creation ---> System.NullReferenceException: Object reference not set to an instance of an object.
            //    at Akka.Actor.Props.NewActor()
            //    at Akka.Actor.ActorCell.CreateNewActorInstance()
            //    at Akka.Actor.ActorCell.<>c__DisplayClass115_0.<NewActor>b__0()
            //    at Akka.Actor.ActorCell.UseThreadContext(Action action)
            //    at Akka.Actor.ActorCell.NewActor()
            //    at Akka.Actor.ActorCell.Create(Exception failure)
            //    --- End of inner exception stack trace ---
            //    at Akka.Actor.ActorCell.Create(Exception failure)
            //    at Akka.Actor.ActorCell.SysMsgInvokeAll(EarliestFirstSystemMessageList messages, Int32 currentState)
            // [WARNING][6/9/2017 10:35:28 PM][Thread 0010][akka://test/user/foo1] DeadLetter from [akka://test/system/testActor1#1379555086] to [akka://test/user/foo1#1059254576]: <Received dead letter from [akka://test/system/testActor1#1379555086]: blah>
            // [INFO][6/9/2017 10:35:28 PM][Thread 0009][akka://test/user/foo1] Message String from akka://test/system/testActor1 to akka://test/user/foo1 was not delivered. 1 dead letters encountered.
        }
    }

    public class FooActor : ReceivePersistentActor
    {
        List<string> stuff = new List<string>();
        public FooActor()
        {
            PersistenceId = Guid.NewGuid().ToString();
            Recover<string>(x =>
            {
                stuff.Add(x);
            });
            Command<string>(x =>
            {
                Persist($"I said: {x}", evt =>
                {
                    stuff.Add(evt);
                });
            });
        }
        public override string PersistenceId { get; }
    }
```

**Comment by gengle** [comment]
I was not able to reproduce with a new project:

- netstandard; net461 framework
- Akka 1.2.0
- Akka.Persistence 1.2.0.36-beta
- Akka.TestKit.NUnit 1.1.3

Attached is test solution for reference.  I'll continue isolating in hopes of finding the culprit.

**Comment by Horusiath** [comment]
@gengle have you tried to add persistence config i.e. 

```csharp
class PersistenceTestFixture : TestKit 
{
    public PersistenceTestFixture():base(config:Persistence.DefaultConfig()) {}
}
```

**Comment by Aaronontheweb** [comment]
closed via #3770
